### PR TITLE
Remove sudo from mo install_prerequisites.sh

### PR DESCRIPTION
**Pull request by dpigasin**
### Tickets:
 - *22208*

### CODE REVIEW

**Review by generalova-kate — APPROVED** [review]

File: model-optimizer/install_prerequisites/install_prerequisites.sh [hunk 1]
```diff
@@ -53,25 +53,13 @@ if [[ $DISTRO == "centos" ]]; then
     elif command -v python3.5 >/dev/null 2>&1; then
         python_binary=python3.5
     fi
-
-    if [ -z "$python_binary" ]; then
-        sudo -E yum install -y https://centos7.iuscommunity.org/ius-release.rpm
```
Comment (generalova-kate) [inline model-optimizer/install_prerequisites/install_prerequisites.sh:58]: did you check the script on centos7?

Comment (dpigasin) [reply model-optimizer/install_prerequisites/install_prerequisites.sh:58]: Yes, I did

**Review by generalova-kate — APPROVED** [review]

**Comment by generalova-kate** [comment]
@lazarevevgeny are you ok with our new settings?

File: model-optimizer/install_prerequisites/install_prerequisites.sh [hunk 2]
```diff
@@ -181,21 +166,20 @@ find_ie_bindings() {
...
 if [[ $V_ENV -eq 1 ]]; then
-    "$python_binary" -m venv "$SCRIPTDIR/../venv${postfix}"
-    source "$SCRIPTDIR/../venv${postfix}/bin/activate"
-    venv_python_binary="$SCRIPTDIR/../venv${postfix}/bin/$python_binary"
+    "$python_binary" -m venv "$HOME/venv_mo${postfix}"
+    source "$HOME/venv_mo${postfix}/bin/activate"
+    venv_python_binary="$HOME/venv_mo${postfix}/bin/$python_binary"
+    # latest pip is needed to install tensorflow
+    "$venv_python_binary" -m pip install --upgrade pip
     $venv_python_binary -m pip install -r "$SCRIPTDIR/../requirements${postfix}.txt"
```
Comment (lazarevevgeny) [inline model-optimizer/install_prerequisites/install_prerequisites.sh:174]: May be wrap `$venv_python_binary` with double quotes? If the path contain spaces then the command will fail

File: model-optimizer/install_prerequisites/install_prerequisites.sh [hunk 3]
```diff
@@ -181,21 +166,20 @@ find_ie_bindings() {
 }
 
 if [[ $V_ENV -eq 1 ]]; then
-    "$python_binary" -m venv "$SCRIPTDIR/../venv${postfix}"
-    source "$SCRIPTDIR/../venv${postfix}/bin/activate"
-    venv_python_binary="$SCRIPTDIR/../venv${postfix}/bin/$python_binary"
+    "$python_binary" -m venv "$HOME/venv_mo${postfix}"
+    source "$HOME/venv_mo${postfix}/bin/activate"
```
Comment (lazarevevgeny) [inline model-optimizer/install_prerequisites/install_prerequisites.sh:170]: Let's introduce additional variable to store `$HOME/venv_mo`. This sub-path is used in 3 places in the code

File: model-optimizer/install_prerequisites/install_prerequisites.sh [hunk 4]
```diff
@@ -181,21 +166,20 @@ find_ie_bindings() {
 }
 
 if [[ $V_ENV -eq 1 ]]; then
-    "$python_binary" -m venv "$SCRIPTDIR/../venv${postfix}"
-    source "$SCRIPTDIR/../venv${postfix}/bin/activate"
-    venv_python_binary="$SCRIPTDIR/../venv${postfix}/bin/$python_binary"
+    "$python_binary" -m venv "$HOME/venv_mo${postfix}"
```
Comment (lazarevevgeny) [inline model-optimizer/install_prerequisites/install_prerequisites.sh:169]: Should we have a single venv directory for all OpenVINO tools? If so then we may need to rename the directory to "venv_openvino" or "venv_ov".

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

**Review by generalova-kate — APPROVED** [review]

**Comment by GlebKazantaev** [comment]
@generalova-kate @dpigasin how do we test this changes?

**Comment by generalova-kate** [comment]
> @generalova-kate @dpigasin how do we test these changes?

It will be tested in docker CI

File: model-optimizer/install_prerequisites/install_prerequisites.bat [hunk 5]
```diff
@@ -119,34 +138,44 @@ IF %errorlevel% NEQ 0 (
...
 :ie_search_end
 
+IF %USE_VENV% == "true" (
+    echo.
+    echo Before running the Model Optimizer, please activate virtualenv environment by running "%VENV_DIR%\Scripts\activate.bat"
+) ELSE (
+    echo.
+    echo [ WARNING ] All Model Optimizer dependencies are installed globally.
+    echo [ WARNING ] If you want to keep Model Optimizer in separate sandbox
+    echo [ WARNING ] run install_prerequisites.sh "{caffe|tf|tf2|mxnet|kaldi|onnx}" venv
```
Comment (rkazants) [inline model-optimizer/install_prerequisites/install_prerequisites.bat:176]: install_prerequisites.sh -> install_prerequisites.bat
Please check that this command working on Windows

File: model-optimizer/install_prerequisites/install_prerequisites.bat [hunk 6]
```diff
@@ -35,10 +37,16 @@ if not "%python_ver%"=="okay" (
 
 :: install Python modules
 
+set USE_VENV="false"
+set VENV_DIR=%USERPROFILE%\Documents\Intel\OpenVINO\venv_openvino
```
Comment (rkazants) [inline model-optimizer/install_prerequisites/install_prerequisites.bat:41]: Is this directory created after OpenVINO installation?

File: model-optimizer/install_prerequisites/install_prerequisites.bat [hunk 7]
```diff
@@ -67,10 +76,20 @@ IF /I "%1%" EQU "" (
...
+    set USE_VENV="true"
+)
+
+IF %USE_VENV% == "true" (
+    python -m venv "%VENV_DIR%"
+    call "%VENV_DIR%\Scripts\activate.bat"
+)
+
+python -m pip install -U pip
+python -m pip install -r "%ROOT_DIR%..\requirements%postfix%.txt"
```
Comment (rkazants) [inline model-optimizer/install_prerequisites/install_prerequisites.bat:89]: It looks that `\` is missed after `%ROOT_DIR%`

Comment (dpigasin) [reply model-optimizer/install_prerequisites/install_prerequisites.bat:89]: ROOT_DIR is equal to `%~dp0` and already ends with `\`

File: model-optimizer/install_prerequisites/install_prerequisites.bat (hunk 6 above)
Comment (dpigasin) [reply model-optimizer/install_prerequisites/install_prerequisites.bat:41]: No, but it also will be created using validation scripts from `deployment_tools/demo` folder, so we can reuse it.

File: model-optimizer/install_prerequisites/install_prerequisites.bat (hunk 5 above)
Comment (dpigasin) [reply model-optimizer/install_prerequisites/install_prerequisites.bat]: The typo has been fixed. The script works fine.

**Review by rkazants — APPROVED** [review]

**Review by azhogov — APPROVED** [review]

**Pull request merged**